moveit_core/kinematics_base/include/moveit/kinematics_base/kinematics_base.h

```diff
@@ -44,6 +44,18 @@
 #include <boost/function.hpp>
 #include <string>
 
+// Import/export for windows dll's and visibility for gcc shared libraries.
```

Comment:
Why do you write this boiler plate manually? There is cmake support for this: https://cmake.org/cmake/help/latest/module/GenerateExportHeader.html

Reply:
This particular code is taken from @seanyen's great work within Microsoft's ms-iot. I originally looked at the `GenerateExportHeader` approach in https://github.com/ros-planning/moveit/pull/2328 but didn't realise how big of a task it would be to fix the whole of moveit. So we just follow Microsoft's lead in this. @traversaro - you are the Microsoft+CMake expert, how much work do you think it would be to convert this code to use the cmake generated export headers instead?

Reply:
I think the generate export header macro could also work. Is there a strong preference to use that?

Reply:
By his wording I expect Robert would prefer the cmake-generated content and so would I if it's not even less readable than the manual version.
@rhaschke do you have an example for using this macro that includes support for `ROS_BUILD_SHARED_LIBS`?

Reply:
I just tried to implement the necessary changes. However, I found this in the URDF project where @wjwwood argues against using this CMake specific function: https://github.com/ros/urdf/issues/11#issuecomment-369365046

Reply:
> @traversaro - you are the Microsoft+CMake expert, how much work do you think it would be to convert this code to use the cmake generated export headers instead?

Sorry, I had missed this comment. I don't think it would be particularly difficult. Furthermore, I think that if you switch to `GenerateExportHeader`, there is no need to use `ROS_BUILD_SHARED_LIBS`, as CMake will generate the code appropriately taking in consideration if the library is compiled as static or as shared, unless I am missing something.